python/lsst/ts/ess/common/device/base_device.py

```diff
@@ -94,6 +94,19 @@ async def __aexit__(
     ) -> None:
         await self.close()
 
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    def __str__(self) -> str:
+        vars_list = [
+            f"{var}={val!r}"
+            for var, val in vars(self).items()
+            if var not in ["log", "terminator"]
+        ]
+        vars_str = ", ".join(vars_list)
```

Comment:
I gently suggest putting the iteration into the join, without the [] so you don't have to actually create the list (instead it creates an iterator when you omit the []):
```
vars_str = ", ".join(
    f"{var}={val!r}"
    for var, val in vars(self).items()
    if var not in {"log", "terminator"}
)
```
Again...a trivial change. I think it might also be just a bit more readable, though that is debatable.